drivers/overlay/overlay.go

```diff
@@ -2162,10 +2162,6 @@ func (d *Driver) getLowerDiffPaths(id string) ([]string, error) {
 // and its parent and returns the size in bytes of the changes
 // relative to its base filesystem directory.
 func (d *Driver) DiffSize(id string, idMappings *idtools.IDMappings, parent string, parentMappings *idtools.IDMappings, mountLabel string) (size int64, err error) {
-	if d.options.mountProgram == "" && (d.useNaiveDiff() || !d.isParent(id, parent)) {
```

Comment:
A random drive-by: Is it really correct to drop the special handling of `!d.isParent`?

Reply:
Thanks!

I think it doesn't matter for the current callers but we probably should do a naive diff if it's not the parent.
@giuseppe WDYT?

Reply:
yes, I also think we should do the naive diff in this case.

Thanks @mtrmac for spotting it